### PR TITLE
REF: refactor/cleanup CSSResolver

### PR DESCRIPTION
**Pull request by ivanovmg**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Refactor ``__call__`` method of ``CSSResolver``.
- Extract methods
- Reorder class attributes
- Add type annotations

### CODE REVIEW

File: pandas/io/formats/excel.py [hunk 1]
```diff
@@ -64,10 +64,11 @@ class CSSToExcelConverter:
     #     without monkey-patching.
 
     def __init__(self, inherited: Optional[str] = None):
+        self.inherited: Optional[Dict[str, str]]
```
Comment (jbrockmendel) [inline pandas/io/formats/excel.py:67]: this goes outside of `__init__`:


```
class Foo:
    inherited : SomeType
    
    def __init__(self, ...):
         self.inherited = ...
      
```

File: pandas/io/formats/css.py [hunk 2]
```diff
@@ -75,101 +130,79 @@ def __call__(self, declarations_str, inherited=None):
         props = dict(self.atomize(self.parse(declarations_str)))
         if inherited is None:
             inherited = {}
+        assert inherited is not None
```
Comment (jreback) [inline pandas/io/formats/css.py:133]: you really need this assert?

File: pandas/io/formats/css.py [hunk 3]
```diff
@@ -75,101 +130,79 @@ def __call__(self, declarations_str, inherited=None):
...
+        props: Dict[str, str],
+        inherited: Dict[str, str],
+    ) -> Dict[str, str]:
         # 1. resolve inherited, initial
         for prop, val in inherited.items():
             if prop not in props:
                 props[prop] = val
 
-        for prop, val in list(props.items()):
+        for prop, val in props.copy().items():
```
Comment (jreback) [inline pandas/io/formats/css.py:149]: don't copy in the interable, rather copy *first* and iterate over the original.

**Review by jreback — CHANGES_REQUESTED** [review]
some comments.

File: pandas/io/formats/css.py (hunk 3 above)
Comment (ivanovmg) [reply pandas/io/formats/css.py]: Changed it. But, out of curiosity, why would it matter?

File: pandas/io/formats/css.py (hunk 2 above)
Comment (ivanovmg) [reply pandas/io/formats/css.py]: No, indeed, that one I forgot. Thank you.

File: pandas/io/formats/excel.py (hunk 1 above)
Comment (ivanovmg) [reply pandas/io/formats/excel.py]: Changed as you suggested.

File: pandas/io/formats/css.py (hunk 3 above)
Comment (jreback) [reply pandas/io/formats/css.py]: doesn't matter, just style-wise its easier to read, e.g. less things then going on in the iteration part

**Review by jreback — APPROVED** [review]

**Review by jreback — APPROVED** [review]
nice!

**Pull request merged**